core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -1805,17 +1909,14 @@ class PlaintextConsumerTest extends BaseConsumerTest {
         s"The current assignment is ${consumer.assignment()}")
   }
 
-  @Test
-  def testConsumingWithNullGroupId(): Unit = {
+  @ParameterizedTest
+  @ValueSource(strings = Array("generic")) // Null group ID only supported for generic group protocol
+  def testConsumingWithNullGroupId(groupProtocol: String): Unit = {
     val topic = "test_topic"
-    val partition = 0;
+    val partition = 0
     val tp = new TopicPartition(topic, partition)
     createTopic(topic, 1, 1)
 
-    TestUtils.waitUntilTrue(() => {
-      this.zkClient.topicExists(topic)
-    }, "Failed to create topic")
-
```

Comment:
Why don't we need this now?

Reply:
I'm not convinced we ever did. Again, this is ZK-specific. The tests work on both variants without this check.

Reply:
I agree with @AndrewJSchofield . Method `createTopics()` already verifies that the topic was created.